src/fileutil.c

```diff
@@ -15,29 +15,94 @@
  */
 
 #include <defines.h>
+#include <errno.h>
+#include <pwd.h>                /* for getpwnam_r() */
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 
 extern char* scenario_path;
 
+int expand_user_path(const char* path, char* expanded_home_path /*The buffer*/, size_t buflen)
+{
+    if (path[0] != '~') {                       /* We have nothing to expand here */
+        return 1;
+    }
+
+    memset(expanded_home_path, '\0', buflen);
+    char* home_dir = NULL;
+
+    if (path[1] == '\0' || path[1] == '/') {                                                    /* '~/path' case */
+        home_dir = getenv("HOME");
+        if (home_dir == NULL) {
+            home_dir = getenv("USERPROFILE");
+        } else {
+            snprintf(expanded_home_path, buflen - 1, "%s%s", home_dir, path + 1);
+        }
+    } else {
+        const char* first_slash = NULL;
+        first_slash = strchr(path, '/');                                                        /* substring starting from '/' */
+        size_t linux_username_limit = 32;                                                       /* As of now */
+        char* username = NULL;
+        if ((first_slash != NULL) && ((first_slash - (path + 1)) <= linux_username_limit)) {    /* '~someuser/blah' case */
```

Comment:
I think you can remove `!= NULL` and the parentheses around `first_slash`.

Reply:
Thanks for suggestion. But I believe it will be optimized by compiler anyway and in my opinion `if (first_slash != NULL)`, or even better `if (NULL != first_slash)`, is more readable than `if (first_slash)`. What do you think? However, If it is a coding guideline for this repo, I will oblige and make changes as you suggest.

Reply:
I don't mind. If you find it more readable, that's fine with me.